lib/client/client.go

```diff
@@ -203,10 +246,156 @@ func (proxy *ProxyClient) ReissueUserCerts(ctx context.Context, params ReissuePa
 	if params.RouteToDatabase.ServiceName != "" {
 		key.DBTLSCerts[params.RouteToDatabase.ServiceName] = certs.TLS
 	}
+	return key, nil
+}
 
-	// save the cert to the local storage (~/.tsh usually):
-	_, err = localAgent.AddKey(key)
-	return trace.Wrap(err)
+// IssueUserCertsWithMFA generates a single-use certificate for the user.
+func (proxy *ProxyClient) IssueUserCertsWithMFA(ctx context.Context, params ReissueParams) (*Key, error) {
+	localAgent := proxy.teleportClient.LocalAgent()
+	key, err := localAgent.GetKey(WithKubeCerts(params.RouteToCluster))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cert, err := key.SSHCert()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	tlsCert, err := key.TeleportTLSCertificate()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	rootClusterName, err := tlsca.ClusterName(tlsCert.Issuer)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if params.RouteToCluster == "" {
+		params.RouteToCluster = rootClusterName
+	}
+
+	// Note: always connect to root for getting new credentials.
+	clt, err := proxy.ConnectToCluster(ctx, params.RouteToCluster, true)
```

Comment:
Shouldn't we use `proxy.ConnectToRootCluster()` here if we only ever want to connect to root? Or is the comment inaccurate?